Generator crashes trying to use `google/cloud/webrisk/v1/webrisk_v1.yaml` file

This command:

```shell
go -C generator/ run ./cmd/sidekick generate -project-root .. -specification-source google/cloud/webrisk/v1 -service-config google/cloud/webrisk/v1/webrisk_v1.yaml -output src/generated/webrisk
```

crashes on:

```

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x30 pc=0x1008ae204]

goroutine 1 [running]:
github.com/googleapis/google-cloud-rust/generator/internal/parser.makeAPIForProtobuf(0x1400013d380, 0x1400045e500)
        /Users/coryan/google-cloud-rust/generator/internal/parser/protobuf.go:227 +0x154
```

I assume the service config does not have a documentation section:

https://github.com/googleapis/google-cloud-rust/blob/eaf3b0d7be3478395a3ea90f60c21b153eae7bfb/generator/internal/parser/protobuf.go#L225-L229

--- a/generator/internal/parser/openapi.go
+++ b/generator/internal/parser/openapi.go
@@ -79,7 +79,9 @@ func makeAPIForOpenAPI(serviceConfig *serviceconfig.Service, model *libopenapi.D
 	if serviceConfig != nil {
 		result.Name = strings.TrimSuffix(serviceConfig.Name, ".googleapis.com")
 		result.Title = serviceConfig.Title
-		result.Description = serviceConfig.Documentation.Summary
+		if serviceConfig.Documentation != nil {
+			result.Description = serviceConfig.Documentation.Summary
+		}
 	}
 
 	// OpenAPI does not define a service name. The service config may provide

--- a/generator/internal/parser/openapi_test.go
+++ b/generator/internal/parser/openapi_test.go
@@ -21,6 +21,8 @@ import (
 	"github.com/google/go-cmp/cmp"
 	"github.com/google/go-cmp/cmp/cmpopts"
 	"github.com/googleapis/google-cloud-rust/generator/internal/api"
+	"google.golang.org/genproto/googleapis/api/serviceconfig"
+	"google.golang.org/protobuf/types/known/apipb"
 )
 
 func TestOpenAPI_AllOf(t *testing.T) {
@@ -791,6 +793,83 @@ func TestOpenAPI_MakeAPI(t *testing.T) {
 			QueryParameters: map[string]bool{},
 		},
 	})
+}
+
+func TestOpenAPI_MakeApiWithServiceConfig(t *testing.T) {
+	contents, err := os.ReadFile("../../testdata/openapi/secretmanager_openapi_v1.json")
+	if err != nil {
+		t.Fatal(err)
+	}
+	model, err := createDocModel(contents)
+	if err != nil {
+		t.Fatal(err)
+	}
+	serviceConfig := &serviceconfig.Service{
+		Name:  "secretmanager.googleapis.com",
+		Title: "Secret Manager API",
+		Apis: []*apipb.Api{
+			{
+				Name: "google.cloud.location.Locations",
+			},
+			{
+				Name: "google.cloud.secretmanager.v1.SecretManagerService",
+			},
+		},
+	}
+	got, err := makeAPIForOpenAPI(serviceConfig, model)
+	if err != nil {
+		t.Fatalf("Error in makeAPI() %q", err)
+	}
+	want := &api.API{
+		Name:        "secretmanager",
+		Title:       "Secret Manager API",
+		Description: "Stores sensitive data such as API keys, passwords, and certificates. Provides convenience while improving security.",
+	}
+
+	if diff := cmp.Diff(got, want, cmpopts.IgnoreFields(api.API{}, "Services", "Messages", "Enums", "State")); diff != "" {
+		t.Errorf("mismatched API attributes (-want, +got):\n%s", diff)
+	}
+
+}
+
+func TestOpenAPI_MakeApiServiceConfigOverridesDescription(t *testing.T) {
+	contents, err := os.ReadFile("../../testdata/openapi/secretmanager_openapi_v1.json")
+	if err != nil {
+		t.Fatal(err)
+	}
+	model, err := createDocModel(contents)
+	if err != nil {
+		t.Fatal(err)
+	}
+	serviceConfig := &serviceconfig.Service{
+		Name:  "secretmanager.googleapis.com",
+		Title: "Secret Manager API",
+		Apis: []*apipb.Api{
+			{
+				Name: "google.cloud.location.Locations",
+			},
+			{
+				Name: "google.cloud.secretmanager.v1.SecretManagerService",
+			},
+		},
+		Documentation: &serviceconfig.Documentation{
+			Summary: "Test Only - Override Description.",
+		},
+	}
+	got, err := makeAPIForOpenAPI(serviceConfig, model)
+	if err != nil {
+		t.Fatalf("Error in makeAPI() %q", err)
+	}
+	want := &api.API{
+		Name:        "secretmanager",
+		Title:       "Secret Manager API",
+		Description: "Test Only - Override Description.",
+	}
+
+	if diff := cmp.Diff(got, want, cmpopts.IgnoreFields(api.API{}, "Services", "Messages", "Enums", "State")); diff != "" {
+		t.Errorf("mismatched API attributes (-want, +got):\n%s", diff)
+	}
+
 }
 
 func TestOpenAPI_SyntheticMessageWithExistingRequest(t *testing.T) {

--- a/generator/internal/parser/protobuf.go
+++ b/generator/internal/parser/protobuf.go
@@ -224,7 +224,9 @@ func makeAPIForProtobuf(serviceConfig *serviceconfig.Service, req *pluginpb.Code
 	}
 	if serviceConfig != nil {
 		result.Title = serviceConfig.Title
-		result.Description = serviceConfig.Documentation.Summary
+		if serviceConfig.Documentation != nil {
+			result.Description = serviceConfig.Documentation.Summary
+		}
 		enabledMixinMethods, mixinFileDesc = loadMixins(serviceConfig)
 		packageName := ""
 		for _, api := range serviceConfig.Apis {

--- a/generator/internal/parser/protobuf_test.go
+++ b/generator/internal/parser/protobuf_test.go
@@ -18,6 +18,7 @@ import (
 	"testing"
 
 	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
 	"github.com/googleapis/google-cloud-rust/generator/internal/api"
 	"google.golang.org/genproto/googleapis/api/annotations"
 	"google.golang.org/genproto/googleapis/api/serviceconfig"
@@ -44,6 +45,23 @@ func TestProtobuf_Info(t *testing.T) {
 	}
 	if diff := cmp.Diff(test.Description, serviceConfig.Documentation.Summary); diff != "" {
 		t.Errorf("description mismatch (-want, +got):\n%s", diff)
+	}
+}
+
+func TestProtobuf_PartialInfo(t *testing.T) {
+	var serviceConfig = &serviceconfig.Service{
+		Name:  "secretmanager.googleapis.com",
+		Title: "Secret Manager API",
+	}
+
+	got := makeAPIForProtobuf(serviceConfig, newTestCodeGeneratorRequest(t, "scalar.proto"))
+	want := &api.API{
+		Name:        "secretmanager",
+		Title:       "Secret Manager API",
+		Description: "",
+	}
+	if diff := cmp.Diff(got, want, cmpopts.IgnoreFields(api.API{}, "Services", "Messages", "Enums", "State")); diff != "" {
+		t.Errorf("mismatched API attributes (-want, +got):\n%s", diff)
 	}
 }
 

